algopy_testing/src/algopy/primitives/uint64.py

```diff
@@ -0,0 +1,735 @@
+from __future__ import annotations
+
+from typing import Self
+
+from algopy.error.uint64 import (
+    UInt64BitShiftInvalidValueError,
+    UInt64BitShiftOverflowError,
+    UInt64InvalidValueError,
+    UInt64OverflowError,
+    UInt64UnderflowError,
+    UInt64ZeroDivisionError,
+)
+from algopy.primitives.constants import MAX_UINT64, MAX_UINT64_BIT_SHIFT
+
+
+class UInt64:
+    """
+    A 64-bit unsigned integer, one of the primary data types on the AVM.
+    """
+
+    value: int  # Raw 'int' value representing the UInt64
+
+    def __init__(self, value: int = 0) -> None:
+        """
+        Initialize a UInt64 with a Python int literal or an int variable
+        declared at the module level.
+
+        Args:
+            value (int, optional): The initial value of the UInt64. Defaults to 0.
+
+        Raises:
+            UInt64InvalidValueError: If the provided value is not an integer.
+            UInt64UnderflowError: If the provided value is negative.
+            UInt64OverflowError: If the provided value exceeds the maximum value for a UInt64.
+        """
```

Comment:
do we need all these docstrings for what amounts to a mock class?

Reply:
I verified that when the implementation and stubs are installed in a venv together that both PyCharm and VSCode will show the stub documentation over the implementation, so docstrings aren't required.